webapp/src/app/explorer/explorer.component.ts

```diff
@@ -1,46 +1,45 @@
 import { Component, OnInit } from "@angular/core";
 import { ActivatedRoute } from "@angular/router";
-import { Contact } from "@c4dt/dynacred";
 import CoinInstance from "@dedis/cothority/byzcoin/contracts/coin-instance";
 import DarcInstance from "@dedis/cothority/byzcoin/contracts/darc-instance";
-import Log from "@dedis/cothority/log";
 import CredentialsInstance from "@dedis/cothority/personhood/credentials-instance";
-import { UserData } from "src/app/user-data.service";
+import { CredentialStructBS } from "dynacred";
+import { ByzCoinService } from "src/app/byz-coin.service";
 
 @Component({
     selector: "app-explorer",
     templateUrl: "./explorer.component.html",
 })
 export class ExplorerComponent implements OnInit {
     kind: string | undefined;
-    contact: Contact | undefined;
+    credStruct: CredentialStructBS | undefined;
     coin: CoinInstance | undefined;
     darc: DarcInstance | undefined;
 
     constructor(
         private readonly route: ActivatedRoute,
-        private readonly uData: UserData,
+        private readonly bcs: ByzCoinService,
     ) {}
 
     async ngOnInit() {
         this.route.paramMap.subscribe(async (params) => {
             const id = Buffer.from(params.get("id"), "hex");
 
-            const p = await this.uData.bc.getProofFromLatest(id);
+            const p = await this.bcs.bc.getProofFromLatest(id);
             switch (p.contractID) {
 
                 case CredentialsInstance.contractID:
-                    this.contact = await Contact.fromByzcoin(this.uData.bc, id);
-                    this.kind = "credential";
+                    this.credStruct = await this.bcs.retrieveCredentialStructBS(id);
+                    this.kind = "credentialObservable.ts";
```

Comment:
that looks weird, it's the only `kind` changed and to a filename

Reply:
I blame IntelliJ...